### ISSUE TITLE
About discrete case

### ISSUE THREAD

**Issue by jzstudent** [shariqiqbal2810/maddpg-pytorch#21]
hi, could you tell me in the discrete case , how to explore the action space, just use gumbel_softmax_sample?  I find the self.exploration is not used?

**Comment by shariqiqbal2810** [comment]
Hi, yes that's correct. The code samples from a Gumbel-Softmax distribution when `explore=True` which will explore the action space. Setting `self.exploration = 0.3` for the discrete case is a remnant of old code where I was doing epsilon-greedy exploration, but it is no longer used.

**Comment by jzstudent** [comment]
Thank you very much! Is Gumbel-Softmax better than epsilon-greedy exploration? I use both and find the result is not good

**Comment by shariqiqbal2810** [comment]
It should be better since it's sampling "better" actions at higher rates, rather than only taking the best or completely random actions which epsilon-greedy does, but you never know with deep RL :P
